packages/devtools/lib/src/memory/memory.dart

```diff
@@ -327,71 +408,55 @@ class MemoryColumnSimple<T> extends Column<T> {
   String getValue(T item) => getter(item);
 }
 
-class MemoryChart extends LineChart<MemoryTracker> {
-  MemoryChart(CoreElement parent) : super(parent, classes: 'perf-chart') {
-    processLabel = parent.add(div(c: 'perf-label'));
-    processLabel.element.style.left = '0';
-
-    heapLabel = parent.add(div(c: 'perf-label'));
-    heapLabel.element.style.right = '0';
+class MemoryChart extends CoreElement {
+  MemoryChart(this._memoryScreen) : super('div') {
+    element.id = memoryGraph;
+    element.style
+      ..alignItems = 'flex-end'
+      ..height = '${chartHeight}px'
+      ..paddingTop = '${topPadding}px';
   }
 
-  CoreElement processLabel;
-  CoreElement heapLabel;
-
-  @override
-  void update(MemoryTracker data) {
-    if (data.samples.isEmpty || dim == null) {
-      // TODO:
-      return;
-    }
+  static const String memoryGraph = 'memory_timeline';
+  static const int chartHeight = 320;
+  static const int topPadding = 0;
 
-    // display the process usage
-    final String rss = '${printMb(data.processRss ?? 0, 0)} MB RSS';
-    processLabel.text = rss;
+  final MemoryScreen _memoryScreen;
+  bool chartCreated = false;
+  MemoryPlotly plotlyChart;
 
-    // display the dart heap usage
-    final String used =
-        '${printMb(data.currentHeap, 1)} of ${printMb(data.heapMax, 1)} MB';
-    heapLabel.text = used;
-
-    // re-render the svg
-
-    // Make the y height large enough for the largest sample,
-    const int tenMB = 1024 * 1024 * 10;
-    final int top = (data.maxHeapData ~/ tenMB) * tenMB + tenMB;
+  void updateFrom(MemoryTracker data) {
+    if (!chartCreated) {
+      plotlyChart = MemoryPlotly(memoryGraph, this)..plotMemory();
+      chartCreated = true;
+    }
 
-    final int width = MemoryTracker.kMaxGraphTime.inMilliseconds;
-    final int right = data.samples.last.time;
+    while (data.samples.isNotEmpty) {
```

Comment:
nit: why isn't this a 
for (HeapSample sample in data.samples) {
...
}
data.samples.clear();

Reply:
Should be.